packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/DbSyncStakePoolSearch.ts

```diff
@@ -0,0 +1,242 @@
+        whereClause.push(where);
+        params.push(...identifierParams);
+      }
+      if (filters.status) {
+        query = addSentenceToQuery(
+          query,
+          `
+          LEFT JOIN pool_retire pr ON 
+            pr.id = (
+              SELECT id
+              FROM pool_retire pr2
+              WHERE pr2.hash_id = ph.id
+              ORDER BY id desc 
+              LIMIT 1
+            )
+          `
+        );
+        whereClause.push(getStatusWhereClause(filters.status, { activeEpoch: 'ph.active_epoch_no' }));
+      }
+      groupByClause = ' GROUP BY ph.id, ph.update_id ORDER BY ph.id DESC';
+    } else if (filters?.status) {
+      query = `${Queries.STATUS_QUERY.WITH_CLAUSE} ${Queries.STATUS_QUERY.SELECT_CLAUSE}`;
+      whereClause.push(getStatusWhereClause(filters.status));
+      if (filters?.identifier) {
+        query = addSentenceToQuery(query, Queries.IDENTIFIER_QUERY.JOIN_CLAUSE.OFFLINE_METADATA);
+        const { where, params: identifierParams } = getIdentifierWhereClause(filters.identifier);
+        whereClause.push(where);
+        params.push(...identifierParams);
+      }
+    } else if (filters?.identifier) {
+      const { where, params: identifierParams } = getIdentifierWhereClause(filters.identifier);
+      query = `
+        ${Queries.IDENTIFIER_QUERY.SELECT_CLAUSE}
+        ${getIdentifierFullJoinClause()}
+        WHERE ${where}
+        `;
+      params.push(...identifierParams);
+    }
+    if (whereClause.length > 0) query = addSentenceToQuery(query, ` WHERE ${whereClause.join(' AND ')}`);
+    query = addSentenceToQuery(query, groupByClause);
+    return { params, query };
+  }
+  public async queryStakePools(options?: StakePoolQueryOptions): Promise<Cardano.StakePool[]> {
+    const { params, query } =
+      options?.filters?._condition === 'or'
+        ? await this.buildOrQuery(options?.filters)
+        : await this.buildAndQuery(options?.filters);
+    this.#logger.debug('About to query pool hashes');
+    const poolUpdates = await this.queryPoolHashes(query, options, params);
+    const hashesIds = poolUpdates.map(({ id }) => id);
+    this.#logger.debug(`${hashesIds.length} pools found`);
+    const updatesIds = poolUpdates.map(({ updateId }) => updateId);
+    const [poolDatas, poolRelays, poolOwners, poolRegistrations, poolRetirements, poolRewards, lastEpoch] =
+      await Promise.all([
+        this.queryPoolData(updatesIds),
+        this.queryPoolRelays(updatesIds),
+        this.queryPoolOwners(hashesIds),
+        this.queryRegistrations(hashesIds),
+        this.queryRetirements(hashesIds),
+        this.queryPoolRewards(hashesIds, options?.rewardsHistoryLimit),
+        this.getLastEpoch()
+      ]);
+    return toCoreStakePool({
+      lastEpoch,
+      poolDatas,
+      poolOwners,
+      poolRegistrations,
+      poolRelays,
+      poolRetirements,
+      poolRewards: poolRewards.filter((pr) => !!pr) as EpochReward[]
```

Comment:
This is fine, but for future reference there is a util in core package which I think is better than explicit casting:
```suggestion
      poolRewards: poolRewards.filter(util.isNotNil)
```